### DISCUSSION TITLE
Anzeige aktueller Betriebmodus Haus-Batterie (Gesperrt, Entladung, Beladung)

### DISCUSSION THREAD

**Discussion by WolfImBusch** [solectrus/solectrus#3828] [Ideas]
Einen schönen guten Morgen,

als "Weihnachtsprojekt" hatte ich mir die Installation von "batcontrol" vorgenommen (https://github.com/muexxl/batcontrol). Sehr grob zusammengefasst: die Software "kümmert" sich um meine Hausbatterie (z.B. prädiktives Laden anhand des Strompreises, Lastprofil, PV-Prognose).

Dazu kann die Batterie drei verschiedene Stati annehmen, die auch über MQTT gesammelt werden könnten ...
MODE 10 - DISCHARGE ALLOWED
MODE 0 - AVOID DISCHARGE
MODE -1 - CHARGE FROM GRID

Ich würde gerne diese Stati in Solectrus darstellen. Es ist vielleicht ein sehr oder zu spezieller Wunsch, aber vielleicht auch für andere Benutzer interessant, wenn andere HEMS mal soweit sind ... :-)

Falls ja .. ich könnte mir da zwei Möglichkeiten vorstellen
1) Änderung des Batterie-Symbols (das Häuschen) je nach Status 

2) Benutzerdefinierte Verwendung des Status-Feldes oben rechts neben der Uhrzeit 
<img width="237" alt="Bildschirmfoto 2024-12-29 um 10 29 01" src="https://github.com/user-attachments/assets/906d393f-62c5-486d-be1a-09490f134827" />

Gesehen habe ich das aktuell nur in der Live-Demo - allerdings nicht in meinem aktuellen Solectrus Dashboard (ich glaube in früheren Versionen wurde hier der Verbindungsstatus dargestellt)
Kann man diese Felder irgendwie zu einer Darstellung "überreden"? Ich vermute, dass diese aktuell SENEC spezifisch sind ...

Herzlichen Dank und viele Grüße

**Comment by ledermann** [comment]
Der Status oben rechts neben der Uhrzeit ist ein Sensor (`INFLUX_SENSOR_SYSTEM_STATUS`), der Zeichenketten verarbeitet. In der Doku wird das hier beschrieben:
https://docs.solectrus.de/referenz/dashboard/konfiguration.html#influx_sensor_system_status

Du könntest also deinen Status mit dem MQTT-Collector abholen und in die InfluxDB schreiben (in ein Measurement/Field deiner Wahl). Dies gibst du dann mit `INFLUX_SENSOR_SYSTEM_STATUS` an, dann wird der Status im UI angezeigt.

In der Live-Demo ist es so, dass die Daten größtenteils aus dem SENEC-Collector kommen. Dieser schreibt auch einen Status (in Measurement `SENEC`, Field `current_state`). Bei mir steht daher in der `.env` das hier:

```
INFLUX_SENSOR_SYSTEM_STATUS=SENEC:current_state
```

Wenn der Sensor nicht angegeben wird, zeigt SOLECTRUS nur "LIVE" oder "Keine Verbindung" an.

**Reply by WolfImBusch** [reply]
Hi @ledermann 

vielen Dank für Deine wie immer sensationell schnelle Antwort. Habe wieder was gelernt. Ich schwöre, ich habe vor dem Stellen der Frage die Doku gelesen ... :-)

Ich denke ich weiß was ich zu tun habe - mit einer Ausnahme ... Der aktuelle Betriebs-Mode wird leider als Wert über MQTT übertragen, also z.B. "10". In Solectrus würde ich natürlich gerne einen Text anzeigen (z.B. "Discharge allowed"). Ich vermute deshalb, dass ich im MQTT Kollektor in Abhänigkeit des Werts (was leider kein JSON Objekt ist) in die InfluxDB einen String schreiben müsste. Also sowas wie

wenn (mode = 10) --> schreibe in InfluxDB "Discharge Allowed".

Würde sowas gehen? Falls nicht - ist nicht tragisch - Du hast bestimmt gerade besseres zu tun ..

Nochmal Danke und viele Grüße!!

**Reply by ledermann** [reply]
😄 

Wenn du den MQTT-Collector auf das Tag `develop` wechselst, kannst du Formeln auch für einfache Werte benutzen (bislang ging das nur für JSON). Das ist noch nicht dokumentiert, daher hier nur der Link zum entsprechenden Commit:
https://github.com/solectrus/mqtt-collector/commit/0a1e8848aa55b1166ec61606c4f0db405ace9778

Du könntest dir damit so etwas basteln:

```
MAPPING_42_FORMULA="IF({value} = 10, 'DISCHARGE ALLOWED', IF({value} = 0, 'AVOID DISCHARGE', IF({value} = -1, 'CHARGE FROM GRID', 'UNKNOWN MODE')))"
```

**Reply by WolfImBusch** [reply]
Vielen, vielen Dank!! Klasse! Und: nen guten Rutsch!!!

edit: 
<img width="268" alt="Bildschirmfoto 2024-12-29 um 12 07 33" src="https://github.com/user-attachments/assets/029291a1-bec1-4834-895a-ae68112263dd" />
Yeah! :-)
